WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -316,43 +332,49 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
-    }
-
-    Class controllerClass;
-    if ([self isGeneralSection:indexPath.section]) {
-        switch (indexPath.row) {
-            case BlogDetailsRowViewSite:
-                [self showViewSiteForBlog:self.blog];
-                break;
-            case BlogDetailsRowViewAdmin:
-                [self showViewAdminForBlog:self.blog];
-                break;
-            case BlogDetailsRowStats:
-                [WPAnalytics track:WPAnalyticsStatStatsAccessed];
-                controllerClass =  [StatsViewController class];
-                break;
-            default:
-                break;
-        }
-    } else if ([self isPublishSection:indexPath.section]) {
-        switch (indexPath.row) {
-            case BlogDetailsRowBlogPosts:
-                [self showPostList];
-                return;
-            case BlogDetailsRowPages:
-                [self showPageList];
-                return;
-            case BlogDetailsRowComments:
-                [WPAnalytics track:WPAnalyticsStatOpenedComments];
-                controllerClass = [CommentsViewController class];
-                break;
-            default:
-                break;
+    Class controllerClass = nil;
+    NSInteger section = [self.tableSections[indexPath.section] integerValue];
+    switch (section) {
+        case (TableViewSectionGeneralType):
+            switch (indexPath.row) {
+                case BlogDetailsRowViewSite:
+                    [self showViewSiteForBlog:self.blog];
+                    break;
+                case BlogDetailsRowViewAdmin:
+                    [self showViewAdminForBlog:self.blog];
+                    break;
+                case BlogDetailsRowStats:
+                    [WPAnalytics track:WPAnalyticsStatStatsAccessed];
+                    controllerClass =  [StatsViewController class];
+                    break;
+                default:
+                    break;
+            }
+            break;
+        case (TableViewSectionPublishType):
+            switch (indexPath.row) {
+                case BlogDetailsRowBlogPosts:
+                    [self showPostList];
+                    return;
+                case BlogDetailsRowPages:
+                    [self showPageList];
+                    return;
+                case BlogDetailsRowComments:
+                    [WPAnalytics track:WPAnalyticsStatOpenedComments];
+                    controllerClass = [CommentsViewController class];
+                    break;
+                default:
+                    break;
+            }
+        break;
+        case (TableViewSectionAppearance):
+            
+            break;
+        case(TableViewSectionConfigurationType):{
+            SiteSettingsViewController *editSiteViewController = [[SiteSettingsViewController alloc] initWithBlog:self.blog];
+            [self.navigationController pushViewController:editSiteViewController animated:YES];
         }
+        break;
```

Comment:
Heads up, i've just updated this exact code section [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/4170/files#diff-2c22ee07f415c85bb9dc440e62c7c008L313) (don't worry, it's in develop!).